src/util/mod.rs

```diff
@@ -683,6 +683,114 @@ pub(crate) unsafe fn copy_unchecked(src: &[u8], dst: &mut [u8]) {
     };
 }
 
+/// Unsafely transmutes the given `src` into a type `Dst`.
+///
+/// # Safety
+///
+/// The value `src` must be a bit-valid instance of `Dst`.
+#[inline(always)]
+pub(crate) const unsafe fn transmute_unchecked<Src, Dst>(src: Src) -> Dst {
+    static_assert!(Src, Dst => core::mem::size_of::<Src>() == core::mem::size_of::<Dst>());
+    #[repr(C)]
+    union Transmute<Src, Dst> {
+        src: ManuallyDrop<Src>,
+        dst: ManuallyDrop<Dst>,
+    }
+    // SAFETY: The caller promises that `src` is a bit-valid instance of `Dst`.
```

Comment:
Also argue that this is a transmute? Namely that `src` and `dst` are at the same byte offsets within `Transmute` and that `ManuallyDrop<T>` has the same layout as `T`.

Reply:
[Done](https://github.com/google/zerocopy/compare/96b54d1b8190c3b88a8d6d0315e14fd241a5a35a..9844c1f976bafef394033070eaf3cef0cf9e4c58#diff-6965df871359c91ee03b5564314f3e131c0877b8603aea626b5d8f9180eb4ee8R700-R717)